sdk/schemaregistry/azure-data-schemaregistry-jsonschema/src/main/java/com/azure/data/schemaregistry/jsonschema/SchemaRegistryJsonSchemaSerializer.java

```diff
@@ -0,0 +1,345 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.schemaregistry.jsonschema;
+
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.exception.ResourceNotFoundException;
+import com.azure.core.http.ContentType;
+import com.azure.core.models.MessageContent;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.serializer.SerializerAdapter;
+import com.azure.core.util.serializer.SerializerEncoding;
+import com.azure.core.util.serializer.TypeReference;
+import com.azure.data.schemaregistry.SchemaRegistryAsyncClient;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+
+import static com.azure.core.util.FluxUtil.monoError;
+
+/**
+ * Class that serializes and deserializes objects using <a href="https://json-schema.org/">JSON schema</a>.
+ *
+ * @see SchemaRegistryJsonSchemaSerializerBuilder
+ */
+public final class SchemaRegistryJsonSchemaSerializer {
```

Comment:
We originally thought about something similar for apache avro serializer but with ObjectSerializer... decided against it because there were too many different methods. I'll ask the feature team and get back to you if we want to do the same thing here.

Reply:
I created a GitHub issue to investigate this: https://github.com/Azure/azure-sdk-for-java/issues/36431